core/controllers/question_editor_test.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the Question Editor controller."""
+
+import json
+
+from core.domain import question_domain
+from core.domain import question_services
+from core.platform import models
+from core.tests import test_utils
+import feconf
+
+(question_models,) = models.Registry.import_models([models.NAMES.question])
+
+
+class EditableQuestionDataHandlerTest(test_utils.GenericTestBase):
+    """Tests put and delete methods of editable questions data handler, get
+    method of questions batch handler, post method question creation handler
+    and get method of question manager handler.
+    """
+
+    def setUp(self):
+        super(EditableQuestionDataHandlerTest, self).setUp()
+
+        self.question_id = 'dummy'
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.set_admins([self.ADMIN_USERNAME])
+
+        self.question = question_domain.Question(
+            self.question_id,
+            self._create_valid_question_data('ABC'),
+            1, 'en', 'private')
+
+    def test_get(self):
+        question_services.add_question(self.admin_id, self.question)
+        with self.swap(feconf, 'ENABLE_NEW_STRUCTURES', True):
+            self.login(self.ADMIN_EMAIL)
+            response_dict = self.get_json('%s/%s' % (
+                feconf.QUESTION_DATA_URL, self.question_id))
+            self.assertEqual(response_dict['username'], 'adm')
```

Comment:
These assert statements will change when the above comment is done.

Reply:
Yes, changed